src/theory/quantifiers/inst_strategy_mbqi.cpp

```diff
@@ -54,6 +56,14 @@ InstStrategyMbqi::InstStrategyMbqi(Env& env,
   }
 }
 
+void InstStrategyMbqi::ppNotifyAssertions(const std::vector<Node>& assertions)
+{
+  for (const Node& a : assertions)
+  {
+    d_notified_assertions.insert(a);
```

Comment:
Instead of storing the assertions, how about we just compute the symbols immediately?

That would mean that you'd have `context::CDHashSet<Node> d_globalSymbols;` only. This method would populate this set using the code you wrote in collectGlobalSymbols.

Reply:
done